### PR TITLE
.NET Core 1.0.0 support

### PR DESCRIPTION
**Pull request by lstratman**
Everything under #433.

### CODE REVIEW

**Comment by brentarias** [comment]
What are the criteria that must be met before this PR (and over a dozen others, such as #437) will be merged?  And what timeline?

**Comment by tjanczuk** [comment]
This PR takes priority over any other given the scope of the change. I hope to review and marge by early next week.

File: README.md [hunk 1]
```diff
@@ -374,13 +374,21 @@ var add7 = edge.func({
...
+
+ 1. It should target the `netcoreapp1.0` framework moniker.
+ 2. It should reference `Microsoft.NETCore.DotNetHost` and `Microsoft.NETCore.DotNetHostPolicy`.  This is required so that the publish process can provide all the native libraries required to create a completely standalone version of your application.
+ 3. `"emitEntryPoint": true` should be present under `buildOptions`.  You can add an empty `Main()` implementation to your project to accomodate it; this method will not be called, but is just a requirement in order for `dotnet publish` to generate a completely standalone app.
+
+On your development machine, you would run `dotnet publish -r [target runtime for your production server]` (i.e. `dotnet publish -r ubuntu.14.04-x64`) to aggregate the package assemblies and native libraries necessary to run your application.  You can copy the contents of the publish directory up to your SDK- and CLI-less server and use them directly in Edge.js by setting the  `EDGE_APP_ROOT` environment variable to the directory on the server that you copied the published application to.
+
```
Comment (tjanczuk) [inline README.md]: @lstratman what would be useful is a "hello, world" samples for .NET Core demonstrating the various options (published app, on the fly compilation).

Comment (lstratman) [inline README.md]: Excellent idea, I will add a samples directory with the various scenarios, each with their own README.md file indicating how to compile and run them.  Might not happen until early next week.

**Comment by tjanczuk** [comment]
@lstratman Is there a reason Edge.js NuGet package does not have a dependency on Edge.js.CSharp, but instead the bootstrap project depends on both?

**Comment by lstratman** [comment]
Since Edge.js.CSharp is only needed for dynamic compilation of C# literals, I didn't want to force that dependency on applications that only reference pre-compiled assemblies.  The bootstrap application, on the other hand, was in place so that applications can get up and running quickly with .NET languages quickly (without a project.json), which generally implied the use of dynamic compilation of literals instead of pre-compiled assemblies.

**Comment by tjanczuk** [comment]
Makes sense, thanks!

**Comment by tjanczuk** [comment]
@lstratman Have you ran the `test\double\double_test` with the Edge.js NuGet package built from the PR bits? This is failing for me with 

```
C:\projects\edge\test\double\double_test\bin\Release>mstest /testcontainer:double_test.dll /noisolation
Microsoft (R) Test Execution Command Line Tool Version 12.0.21005.1
Copyright (c) Microsoft Corporation. All rights reserved.

Loading double_test.dll...
Starting execution...

Unhandled Exception: System.AccessViolationException: Attempted to read or write protected memory. This is often an indi
cation that other memory is corrupt.
   at EdgeJs.Edge.NodeStart(Int32 argc, String[] argv)
   at EdgeJs.Edge.<>c.<Func>b__10_0()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object
state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, B
oolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

**Comment by tjanczuk** [comment]
@lstratman never mind, found it. The entry point to start@node in node.dll has changed, probably as a result of my updating the compiler (since I did not change node version).

**Comment by tjanczuk** [comment]
https://www.nuget.org/packages/Edge.js/5.9.2

**Comment by tjanczuk** [comment]
@lstratman did you use `test\testall.bat` on windows? It is failing for me for all but the most recently built version.

**Comment by tjanczuk** [comment]
@lstratman figured it out, never mind.

**Comment by tjanczuk** [comment]
This has been huge work @lstratman, thank you! All bits are now released /cc @ChristianWeyer 

Edge.js 5.9.2 on Nuget: https://www.nuget.org/packages/Edge.js/5.9.2
Edge.js.CSharp 1.0.0 on Nuget: https://www.nuget.org/packages/Edge.js.CSharp/
edge@5.9.2 on NPM: https://www.npmjs.com/package/edge

I will be out for a week now, I will do some promo around it when back. Maybe I will update to Node 6.x before. 

Again, fantastic job, this would not have happened without you @lstratman!

**Comment by glennblock** [comment]
👍👍👍👍

**Comment by pauldotknopf** [comment]
I notice the README still says "coming soon" for CoreCLR. Is that just out-dated?

**Pull request merged**